### PR TITLE
Create _repr_html_ for viewing data heavy classes in Jupyter

### PR DESCRIPTION
**Pull request by jp-dark**
Create `_repr_html_` methods for the `GroupSchema`, `DataspaceCreator`, and `NetCDF4ConverterEngine` classes.

### CODE REVIEW

File: tiledb/cf/core.py [hunk 1]
```diff
@@ -19,6 +19,53 @@
...
+            f"={schema.allows_duplicates}</td></tr>\n"
+        )
+    output.write(
+        f'<tr><td style="text-align: left">coords_filters={schema.coords_filters}'
+        f"</td>\n"
+    )
+    output.write("</table>\n")
+    output.write("</li>\n")
+    output.write("</ul>\n")
+    return output.getvalue()
```
Comment (gsakkis) [inline tiledb/cf/core.py:66]: Might be an overkill here but people often use a templating engine such as [Jinja2](https://github.com/pallets/jinja) or [Mako](https://www.makotemplates.org/) for non-trivial text generation. There are also specialized HTML generating libraries such as [yattag](https://www.yattag.org/) that guarantee well-formed output (so no need to test it with tidylib).

File: tiledb/cf/creator.py [hunk 2]
```diff
@@ -992,6 +1068,19 @@ def __repr__(self):
...
         )
 
+    def html_summary(self) -> str:
+        """Returns a string HTML summary of the :class:`AttrCreator`."""
+        filters_str = ""
+        if self.filters:
+            filters_str = ", filters=FilterList(["
+            for attr_filter in self.filters:
+                filters_str += repr(attr_filter) + ", "
+            filters_str += "])"
```
Comment (gsakkis) [inline tiledb/cf/creator.py]: Isn't this equivalent to `filters_str = f", filters=FilterList({self.filters})"`? (bonus: doesn't include a trailing comma)

File: tiledb/cf/creator.py [hunk 3]
```diff
@@ -1042,6 +1131,19 @@ def domain(self) -> Tuple[Optional[DType], Optional[DType]]:
...
         return self.base.domain
 
+    def html_summary(self) -> str:
+        """Returns a string HTML summary of the :class:`DimCreator`."""
+        filters_str = ""
+        if self.filters:
+            filters_str = ", filters=FilterList(["
+            for dim_filter in self.filters:
+                filters_str += repr(dim_filter) + ", "
+            filters_str += "])"
```
Comment (gsakkis) [inline tiledb/cf/creator.py:1131]: Same comment as above

File: tests/engines/netcdf4_engine/test_netcdf4_converter_engine.py [hunk 4]
```diff
@@ -77,6 +77,16 @@ def test_converter_from_netcdf(self, netcdf_file, tmpdir, collect_attrs):
         converter.convert_to_group(uri)
         self.check_attrs(uri)
 
+    def test_converter(self, netcdf_file):
```
Comment (gsakkis) [inline tests/engines/netcdf4_engine/test_netcdf4_converter_engine.py:80]: `test_repr_html`?

File: tiledb/cf/engines/netcdf4_engine.py [hunk 5]
```diff
@@ -60,10 +60,14 @@ class NetCDFCoordToDimConverter(SharedDim, NetCDFDimConverter):
 
     def __repr__(self):
         return (
-            f"Variable(name={self.input_name}, dtype={self.input_dtype}) -> "
+            f"NetCDFVariable(name={self.input_name}, dtype={self.input_dtype}) -> "
```
Comment (gsakkis) [inline tiledb/cf/engines/netcdf4_engine.py:63]: Is this deliberately the same with the respective `NetCDFVariableConverter` repr?

**Review by gsakkis — APPROVED** [review]

File: tiledb/cf/core.py (hunk 1 above)
Comment (jp-dark) [reply tiledb/cf/core.py:66]: I leave it is for now, but I'll consider yattag or similar. Especially if it gets any more complicated than it currently is.

File: tiledb/cf/creator.py (hunk 2 above)
Comment (jp-dark) [reply tiledb/cf/creator.py]: Technically, I think this requires me to bump the required TileDB-Py version. I'm doing that soon anyway though.

File: tiledb/cf/engines/netcdf4_engine.py (hunk 5 above)
Comment (jp-dark) [reply tiledb/cf/engines/netcdf4_engine.py:63]: Yes, they map from the same thing (a NetCDF variable) to a new TileDB object that is defined by the super class.

**Pull request merged**